reaktive/src/commonTest/kotlin/com/badoo/reaktive/observable/RefCountTest.kt

```diff
@@ -111,6 +113,18 @@ class RefCountTest {
         assertFalse(disposable.isDisposed)
     }
 
+    // @Test
+    // fun does_not_disconnect_from_upstream_WHEN_subscriberCount_is_1_and_first_or_error_with_replay() {
+    //     val disposable = Disposable()
+    //     val upstream = PublishSubject<Int>()
+    //     val refCount = upstream.replay(1).refCount(subscriberCount = 1)
+    //     refCount.test()
+    //     upstream.onNext(1)
+    //     refCount.firstOrError().test()
+    //
+    //     assertFalse(disposable.isDisposed)
+    // }
+
```

Comment:
```suggestion
    @Test
    fun does_not_disconnect_WHEN_connected_and_new_observer_disposed_synchronously_on_emission() {
        val disposable = Disposable()

        val upstream =
            testUpstream(
                connect = { onConnect -> onConnect?.invoke(disposable) },
                subscribe = { observer ->
                    observer.onSubscribe(Disposable())
                    observer.onNext(1)
                }
            )

        val refCount = upstream.refCount(subscriberCount = 1)

        refCount.test() // First subscription

        refCount.subscribe(
            object : DefaultObservableObserver<Int?> {
                private var disposableRef: Disposable? by AtomicReference(null)

                override fun onSubscribe(disposable: Disposable) {
                    disposableRef = disposable
                }

                override fun onNext(value: Int?) {
                    requireNotNull(disposableRef).dispose()
                }
            }
        )

        assertFalse(disposable.isDisposed)
    }
```

Reply:
Suggestion implemented, though I couldn't use AtomicReference as a delgate so I used it directly.

Reply:
This is fine, thanks!